### ISSUE TITLE
mypy doesnt raise error if boolean is used with Union inside the List

### ISSUE THREAD

**Issue by anshulpatel25** [python/mypy#9223]
* Are you reporting a bug, or opening a feature request? Bug
* Please insert below the code you are checking with mypy
## Code

```python
from typing import List, Union


my_list: List[Union[int, float]] = [1,2.0,3.0,True,False]
```

* What is the actual behavior/output?
`List[Union[int, float]]` should only allow integer and float values in the list, but somehow mypy didn't raise an error when I added boolean to the list.
```bash
(venv) ➜ mypy main.py
Success: no issues found in 1 source file
```
* What is the behavior/output you expect?
`List[Union[int, float]]` should only allow integer and float values in the list

* What are the versions of mypy and Python you are using?
## Python Version

```bash
(venv) ➜ python --version
Python 3.8.2
```
## Mypy Version

```bash
(venv) ➜  pip freeze
mypy==0.782
mypy-extensions==0.4.3
typed-ast==1.4.1
typing-extensions==3.7.4.2
```
  
* What are the mypy flags you are using? (For example --strict-optional)

I am using no flags.

**Comment by JelleZijlstra** [comment]
That's because bools are ints. You get an error if you use a type that isn't a subtype of int, like str.

**Comment by anshulpatel25** [comment]
Thanks @JelleZijlstra !
